File: backend/src/main/java/coursepick/coursepick/domain/course/CourseRepository.java
package coursepick.coursepick.domain.course;

import org.springframework.data.domain.Slice;

import java.util.List;
import java.util.Optional;

public interface CourseRepository {

    Course save(Course course);

    void saveAll(Iterable<? extends Course> courses);

    Slice<Course> findAllHasDistanceWithin(CourseFindCondition condition);

    List<Course> findByIdIn(List<String> ids);

    Optional<Course> findById(String id);

    Optional<Course> findByName(CourseName courseName);

    void delete(Course course);
}

File: android/app/src/main/java/io/coursepick/coursepick/domain/CourseRepository.kt
package io.coursepick.coursepick.domain

interface CourseRepository {
    suspend fun courses(
        mapCoordinate: Coordinate,
        userCoordinate: Coordinate? = null,
    ): List<Course>

    suspend fun nearestCoordinate(
        selected: Course,
        current: Coordinate,
    ): Coordinate
}
